dbft.go

```diff
@@ -61,6 +61,8 @@ func (d *DBFT[H]) addTransaction(tx Transaction[H]) {
 			return
 		}
 
+		d.verifyPreCommitPayloadsAgainstPreBlock()
```

Comment:
> Why not before `d.createAndCheckBlock()`?

Because it's useless to spend time validating PreCommits against invalid block.

> since checkPrepare() should end up the same way.

`checkPrepare()` doesn't perform this verification one more time. PreCommit payload is validated either once it's received (if all transactions are collected) or when all transactions are collected in `addTransaction` callback. So to me there's no way to end up with invalid PreCommit payloads by the moment of PreBlock processing.